modules/Multimode-Networks/src/main/java/cz/cvut/fit/gephi/multimode/GraphColoring.java

```diff
@@ -0,0 +1,123 @@
+package cz.cvut.fit.gephi.multimode;
+
+import java.util.ArrayList;
+
+import org.gephi.graph.api.Column;
+import org.gephi.graph.api.Origin;
+import org.gephi.graph.api.Table;
+import org.gephi.graph.api.Graph;
+import org.gephi.graph.api.GraphController;
+import org.gephi.graph.api.Node;
+import org.openide.util.Lookup;
+
+/**
+ *
+ * @author Jaroslav Kuchar
+ */
+public class GraphColoring {
+
+    private Graph graph;
+    private Node[] nodes;
+
+    public GraphColoring() {
+        // load all possible values of attribute
+        GraphController graphController = Lookup.getDefault().lookup(GraphController.class);
+        graph = graphController.getGraphModel().getGraphVisible();
+        nodes = graph.getNodes().toArray();
+    }
+
+    /**
+     * graph coloring for bipartite networks Implementation inspired by:
+     * http://pages.cs.wisc.edu/~elgar/cs577/Bipartite/Bipartite.java
+     *
+     * @return true if bipartite
+     */
+    public boolean bipartite() {
+        String colorColumnName = "NodeColor-Multimode";
+        Table nodeTable = graph.getModel().getNodeTable();
+        Column colorColumn;
+        
+        if (!nodeTable.hasColumn(colorColumnName)) {
+            colorColumn = nodeTable.addColumn(
+                    colorColumnName,
+                    "Node Color Multimode",
+                    String.class,
+                    Origin.DATA,
+                    NodeColor.BLACK.getValue(),
+                    false);
```

Comment:
Indexed (false) should be true here. In Gephi we create all columns with indexed = true so they can be used in filters and appeareance.

The non-indexed option is for other possible usages of graphstore API outside of Gephi.